webapp/packages/game/src/deck.ts

```diff
@@ -0,0 +1,40 @@
+import { OpenStarTerVillageType } from './types';
+
+type Deck<T> = OpenStarTerVillageType.State.Deck<T>;
+
+export class CardDeck<T> implements Deck<T> {
+  public drawPile: T[];
+  public discardPile: T[];
+  constructor(drawPile?: T[], discardPile?: T[]) {
+    this.drawPile = drawPile || [];
+    this.discardPile = discardPile || [];
+  }
+}
+export interface IDeck {
+  ShuffleDrawPile<T>(deck: Deck<T>, shuffler: (pile: T[]) => T[]): void;
+  ShuffleDiscardPile<T>(deck: Deck<T>, shuffler: (pile: T[]) => T[]): void;
+  PutDiscardPileToDrawPile<T>(deck: Deck<T>): void;
+  Draw<T>(deck: Deck<T>, n: number): T[];
+  Discard<T>(deck: Deck<T>, ts: T[]): void;
+}
```

Comment:
`Deck<T>` / `Deck` / `IDeck` are mixed up in this snippet .
I think we can rename them like `OpenStarTerVillageType.State.Deck<T> ` -> `StateDeck` or whatever would be better than adding a comment in the PR.
![Notification_Center](https://user-images.githubusercontent.com/2024543/146649075-19f6941c-9d5b-4dd9-9194-b0ef9564540a.png)

Reply:
> Minor: I think we can avoid fix/refactor commits of an add feature commit in a PR in the future.

I left those commits for recording how and why my idea changes for code review. I am happy to recommit them after code review is done.

Reply:
> I think `Deck<T>` / `Deck` / `IDeck` are mixed up in this snippet . Maybe we can rename them like `OpenStarTerVillageType.State.Deck<T> ` -> `StateDeck` or whatever and that would be better than adding a comment in the PR. ![Notification_Center](https://user-images.githubusercontent.com/2024543/146649075-19f6941c-9d5b-4dd9-9194-b0ef9564540a.png)

I agree with you. The situation you mentioned blocked me but I could not find a proper solution.
Rename the State.Deck to be State.StateDeck may be the optimal way so far.

Reply:
<img width="869" alt="Screen Shot 2021-12-21 at 10 58 04 pm" src="https://user-images.githubusercontent.com/1487883/146960630-e4f5662a-d763-4c63-b56e-111db9d2a985.png">

The type Deck is renamed as StateDeck under deck-helper scope. I think this is a better approach to avoid re-prefixing the Deck in State scope.